### ISSUE TITLE
CompactionJob Prepare() wrong when boundaries size == 1

### ISSUE THREAD

**Issue by DWAE86** [facebook/rocksdb#11476]
I'm reading the code in compaction_job.cc and wondering if this is wrong
```cpp
void CompactionJob::Prepare() {
...
  if (boundaries_.size() > 1) {
    for (size_t i = 0; i <= boundaries_.size(); i++) {
       ...
    }
    RecordInHistogram(stats_, NUM_SUBCOMPACTIONS_SCHEDULED,
                      compact_->sub_compact_states.size());
  } else {
    compact_->sub_compact_states.emplace_back(c, std::nullopt, std::nullopt,
                                              /*sub_job_id*/ 0);
  }
```
when `num_actual_subcompactions == 2`, boundaries size will be 1, and sub compaction does not work.
if it should be `boundaries_.size() >= 1` ?

**Comment by cbi42** [comment]
Hi - thanks for reporting. You are right, it should be `boundaries_.size() >= 1`.

**Comment by cbi42** [comment]
Fixed in #11501.
